plugin/ochttp/propagation/tracecontext/propagation.go

```diff
@@ -87,15 +90,71 @@ func (f *HTTPFormat) SpanContextFromRequest(req *http.Request) (sc trace.SpanCon
 		return trace.SpanContext{}, false
 	}
 
+	// Extract Tracestate
+	tracestate, ok := tracestateFromRequest(req)
+	if ok == false {
+		return trace.SpanContext{}, false
+	}
+	sc.Tracestate = tracestate
 	return sc, true
 }
 
-// SpanContextToRequest modifies the given request to include a header.
+func tracestateFromRequest(req *http.Request) (*tracestate.Tracestate, bool) {
```

Comment:
I have a general comment: instead of putting the header parsing logic in ochttp, can we have a tracestateFromHttpHeader that can be used by other HTTP propagators?

Reply:
where should it move to?

Reply:
I don't have the exact answer for Go.
For Python we have https://github.com/census-instrumentation/opencensus-python/tree/master/opencensus/trace/propagation.

Reply:
I have moved it out of ochttp.